disease_gene/disease_associates_gene/generative_model_experiments/data/label_functions/disease_gene_lf.py

```diff
@@ -0,0 +1,812 @@
+    "unresponsive", "under-expression", "underexpresed", "down-regulated", "downregulated", "knockdown",
+    "suppressed", "negative", "weak", "lower"
+}
+
+disease_sample_indicators = {
+    "tissue", "cell", "patient", "tumor", "cancer", "carcinoma",
+    "cell line", "cell-line", "group", "blood", "sera", "serum", "fluid", "subset", 
+    "case", "men", "womdn"
+}
+
+diagnosis_indicators = {
+    "prognostic significance of", "prognostic indicator for", "prognostic cyosolic factor",
+    "prognostic parameter for", "prognostic information for", "predict(or|ive) of",
+    "predictor of prognosis in", "indicative of", "diagnosis of", "was positive for",
+    "detection of", "determined by", "diagnositic sensitivity", "dianostic specificity",
+    "prognostic factor", "variable for the identification", "potential therapeutic agent",
+    "prognostic parameter for", "identification of"
+}
+
+no_direct_association = {
+    "not significant", "not significantly", "no association", "not associated",
+    "no correlation between" "no correlation in", "no correlation with", "not correlated with",
+     "not detected in", "not been observed", "not appear to be related to", "neither", 
+     "provide evidence against", "not a constant", "not predictive", "nor were they correlated with"
+    }
+
+weak_association = {
+    "not necessarily indicate", "the possibility", "low correlation", "may be.* important", "might facillitate",
+    "might be closely related to"
+}
+
+negative_indication =  {
+    "failed", "poorly"
+}
+
+method_indication = {
+    "investigated (the effect of|in)", "was assessed by", "assessed", 
+    "compared to", "were analyzed", "evaluated in", "examination of", "examined in",
+    "quantified in" "quantification by", "we review", "(were|was) measured", "we(re)? studied", 
+    "we measured", "derived from", "(are|is) discussed", "to measure", "to study",
+    "to explore", "detection of", "authors summarize", "responsiveness of",
+    "used alone", "blunting of", "measurement of", "detection of", "occurence of", 
+    "our objective was", "to test the hypothesis", "studied in", "were reviewed",
+    "randomized study", "this report considers", "was administered", "determinations of",
+    "we examine(d)?", "(was|we|were) evaluated", "to establish", "were selected", "authors determined",
+    "we investigated", "to assess", "analyses were done", "useful tool for the study of", r"^The effect of",
+    "OBJECTIVE:", "PURPOSE:", "METHODS:", "were applied", "EXPERIMENTAL DESIGN:"
+}
+
+title_indication = {
+    "Effect of", "Evaluation of", 
+    "Clincal value of", "Extraction of",
+    "Responsiveness of", "The potential for",
+    "as defined by immunohistochemistry", "Comparison between",
+    "Characterization of", "A case of", "Occurrence of",
+    "Inborn", "Episodic", "Detection of", "Immunostaining of",
+    "Mutational analysis of"
+}
+
+def LF_DG_IS_BIOMARKER(c):
+    """
```

Comment:
So here you are searching for a sentence that has the format: "<gene> is <phrase from biomarker_indicator> <disease>"?

You are not allowing filler words in between the gene, phrase and disease in this case?  I assume that wouldn't make sense.  Not too familiar with string manipulations though.

Reply:
This is a case of bad nomenclature. I named this function to basically answer the question is the candidate gene a biomarker -> hence IS_BIOMARKER. I agree that i should remove the IS part to avoid confusion.